apps/remote_control/lib/lexical/remote_control/code_intelligence/entity.ex

```diff
@@ -21,15 +22,17 @@ defmodule Lexical.RemoteControl.CodeIntelligence.Entity do
 
   Returns `{:ok, resolved, range}` if successful, `{:error, error}` otherwise.
   """
-  @spec resolve(Document.t(), Position.t()) :: {:ok, resolved, Range.t()} | {:error, term()}
-  def resolve(%Document{} = document, %Position{} = position) do
-    with {:ok, surround_context} <- Ast.surround_context(document, position),
-         {:ok, resolved, {begin_pos, end_pos}} <- resolve(surround_context, document, position) do
+  @spec resolve(Analysis.t(), Position.t()) :: {:ok, resolved, Range.t()} | {:error, term()}
+  def resolve(%Analysis{} = analysis, %Position{} = position) do
+    analysis = Ast.analyze_to(analysis, position)
```

Comment:
I'm not sure this should be done here. I think this should probably be done before the call to `resolve` is made so that we can ensure `analyze_to` is only called once per request.

Reply:
Nevermind; it's fine for this to happen here because this is just a no-op if `analyze_to` was called earlier.